reflect/src/main/java/dagger/reflect/BindingGraph.java

```diff
@@ -44,7 +45,11 @@ private BindingGraph(Map<Key, Binding<?>> bindings, JustInTimeProvider jitProvid
 
     for (int i = 0; i < dependencyKeys.length; i++) {
       Key dependencyKey = dependencyKeys[i];
-      Binding<?> dependency = locateBinding(dependencyKey);
+      @Nullable Binding<?> dependency = locateBinding(dependencyKey);
```

Comment:
This nullable is inferred by `ConstantConditions`, and overriding is could be misleading:
Imagine during maintenance a change to `locateBinding` so that it's `@NonNull`, this would declare it as Nullable still, but the below `dependency == null` will become unnecessary.